go/tools/releasing/cmd/prerelease.go

```diff
@@ -0,0 +1,309 @@
+
+	cmd := exec.Command("make", "lint")
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("'make lint' failed: %v (%v)", string(output), err)
+	}
+
+	return nil
+}
+
+func (p prerelease) commitChanges(skipMake bool) error {
+	commitMessage := "Prepare for releasing " + p.newVersion
+
+	// add changes to git
+	cmd := exec.Command("git", "add", ".")
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("'git add .' failed: %v (%v)", string(output), err)
+	}
+
+	// make ci
+	if skipMake {
+		fmt.Println("Skipping 'make ci'...")
+	} else {
+		fmt.Println("Running 'make ci'...")
+		cmd = exec.Command("make", "ci")
+		if output, err := cmd.CombinedOutput(); err != nil {
+			return fmt.Errorf("'make ci' failed: %v (%v)", string(output), err)
+		}
+	}
+
+	// commit changes to git
+	fmt.Printf("Commit changes to git with message '%v'...\n", commitMessage)
+	cmd = exec.Command("git", "commit", "-m", commitMessage)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("git commit failed: %v (%v)", string(output), err)
+	}
+
+	cmd = exec.Command("git", "log", `--pretty=format:"%h"`, "-1")
+	output, err := cmd.Output()
+	if err != nil {
+		fmt.Println("WARNING: could not automatically get last commit hash.")
+	}
+
+	fmt.Println("Commit successful. Hash of commit:")
+	os.Stdout.Write(output)
+
+	return nil
+}
+
+// updateGoModVersions reads the fromFile (a go.mod file), replaces versions
+// for all specified modules in newModPaths, and writes the new go.mod to the toFile file.
+func (p prerelease) updateGoModVersions(modFilePath tools.ModuleFilePath) error {
+	newGoModFile, err := ioutil.ReadFile(string(modFilePath))
+	if err != nil {
+		panic(err)
+	}
+
+	for _, modPath := range p.modPaths {
+		oldVersionRegex := filePathToRegex(string(modPath)) + tools.SemverRegex
+		r, err := regexp.Compile(oldVersionRegex)
+		if err != nil {
+			return fmt.Errorf("error compiling regex: %v", err)
+		}
+
+		newModVersionString := string(modPath) + " " + p.newVersion
+
+		newGoModFile = r.ReplaceAll(newGoModFile, []byte(newModVersionString))
+	}
+
+	// once all module versions have been updated, overwrite the go.mod file
+	ioutil.WriteFile(string(modFilePath), newGoModFile, 0644)
+
+	return nil
+}
+
+// updateAllGoModFiles updates ALL modules' requires sections to use the newVersion number
+// for the modules given in newModPaths.
+func (p prerelease) updateAllGoModFiles() error {
+	fmt.Println("Updating all module versions in go.mod files...")
+	for _, modFilePath := range p.ModPathMap {
+		if err := p.updateGoModVersions(modFilePath); err != nil {
+			return fmt.Errorf("could not update module versions in file %v: %v", modFilePath, err)
+		}
+	}
+	return nil
+}
+
+func filePathToRegex(fpath string) string {
+	replacedSlashes := strings.Replace(fpath, string(filepath.Separator), `\/`, -1)
+	replacedPeriods := strings.Replace(replacedSlashes, ".", `\.`, -1)
+	return replacedPeriods
```

Comment:
```suggestion
	return regexp.QuoteMeta(fpath)
```

This is probably safer since there may be other meta characters.

Reply:
Good idea; I tried to use the `QuoteMeta` func but found that it did not replace forward slashes (like the ones that exist in the import paths). Perhaps I can do something like this instead:

```suggestion
	quotedMeta := regexp.QuoteMeta(fpath)
	replacedSlashes := strings.Replace(quotedMeta, string(filepath.Separator), `\/`, -1)
	return replacedSlashes
```